### ISSUE TITLE
An error occurs when importing embedchain with gunicorn.

### ISSUE THREAD

**Issue by sasaki000** [embedchain/embedchain#1055]
### 🐛 Describe the bug

An error occurs when importing embedchain with gunicorn.

The following error occurred immediately after starting the flask app.
Perhaps when gunicorn launches multiple threads,
We suspect that the following error occurred because another thread interrupted while creating the folder after checking the folder's existence in client.py.
os.makedirs('data', exist_ok=True)
I think it can be resolved by specifying the exeist_ok option.

- Error(stack trace)
```
[2023-12-23 19:06:19 JST] <ERROR> [orcas]setup(__init__.py:178) - [Errno 17] File exists: '/root/.embedchain'
Traceback (most recent call last):
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/__init__.py", line 168, in setup
    setup_django()
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/__init__.py", line 136, in setup_django
    django.setup()
  File "/usr/local/lib/python3.10/dist-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.10/dist-packages/django/apps/registry.py", line 116, in populate
    app_config.import_models()
  File "/usr/local/lib/python3.10/dist-packages/django/apps/config.py", line 269, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/models/__init__.py", line 19, in <module>
    from orcas import utils, fin
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/utils/__init__.py", line 1297, in <module>
    from .open_ai import client
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/utils/open_ai.py", line 32, in <module>
    from embedchain import App
  File "/usr/local/lib/python3.10/dist-packages/embedchain/__init__.py", line 7, in <module>
    from embedchain.pipeline import Pipeline  # noqa: F401
  File "/usr/local/lib/python3.10/dist-packages/embedchain/pipeline.py", line 28, in <module>
    Client.setup_dir()
  File "/usr/local/lib/python3.10/dist-packages/embedchain/client.py", line 43, in setup_dir
    os.makedirs(CONFIG_DIR)
  File "/usr/lib/python3.10/os.py", line 225, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/.embedchain'
[2023-12-23 19:06:19 JST] <ERROR> [orcas]setup(__init__.py:178) - [Errno 17] File exists: '/root/.embedchain'
Traceback (most recent call last):
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/__init__.py", line 168, in setup
    setup_django()
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/__init__.py", line 136, in setup_django
    django.setup()
  File "/usr/local/lib/python3.10/dist-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.10/dist-packages/django/apps/registry.py", line 116, in populate
    app_config.import_models()
  File "/usr/local/lib/python3.10/dist-packages/django/apps/config.py", line 269, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/models/__init__.py", line 19, in <module>
    from orcas import utils, fin
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/utils/__init__.py", line 1297, in <module>
    from .open_ai import client
  File "/var/workspace/orcas_proj/jupyter_proj/orcas/utils/open_ai.py", line 32, in <module>
    from embedchain import App
  File "/usr/local/lib/python3.10/dist-packages/embedchain/__init__.py", line 7, in <module>
    from embedchain.pipeline import Pipeline  # noqa: F401
  File "/usr/local/lib/python3.10/dist-packages/embedchain/pipeline.py", line 28, in <module>
    Client.setup_dir()
  File "/usr/local/lib/python3.10/dist-packages/embedchain/client.py", line 43, in setup_dir
    os.makedirs(CONFIG_DIR)
  File "/usr/lib/python3.10/os.py", line 225, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/.embedchain'
```

- Version info
```
root@607a27f8ad7e:/var/workspace/orcas_proj/flask_proj# pip3 show embedchain
Name: embedchain
Version: 0.1.42
Summary: Data platform for LLMs - Load, index, retrieve and sync any unstructured data
Home-page:
Author: Taranjeet Singh
Author-email: taranjeet@embedchain.ai
License: Apache License
Location: /usr/local/lib/python3.10/dist-packages
Requires: beautifulsoup4, chromadb, langchain, llama-index, openai, posthog, pypdf, python-dotenv, rich, schema, tiktoken
Required-by:
```

**Comment by deshraj** [comment]
Thanks for creating the issue @sasaki000 and it was a good catch. I have put up a PR to fix this issue. Will be fixed in the 0.1.43 release version

**Comment by taranjeet** [comment]
@sasaki000 : We have fixed the issue. Would love your feedback.
Please feel free to [join our slack](https://embedchain.ai/slack) or [block some time](https://cal.com/taranjeetio/ec) with us in case the issue still exists or you have feedback/feedback requests
